### PR TITLE
[Laravel-5.0] Add the support for Laravel 5 paginator

### PR DESCRIPTION
**Pull request by maximebeaudoin**
Related to the issue https://github.com/thephpleague/fractal/issues/150

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
This doesn't work on php 5.3. The dependencies don't install.

**Comment by maximebeaudoin** [comment]
@GrahamCampbell I know that, it's because `illumintate/contracts` need php 5.4 if you check at https://github.com/illuminate/contracts/blob/master/composer.json#L12. Unfortunately, we need the `Illuminate\Contracts\Pagination\LengthAwarePaginator` if we want this to work with laravel 5.

**Comment by GrahamCampbell** [comment]
Composer has a flag you can run it with to get around this you know. :)

**Comment by GrahamCampbell** [comment]
Edit the travis file to run composer with `--ignore-platform-reqs`.

**Comment by maximebeaudoin** [comment]
Good to know, should i replace `travis_retry composer install --no-interaction --prefer-source --dev` by this `travis_retry composer install --no-interaction --prefer-source --ignore-platform-reqs --dev` from the file https://github.com/thephpleague/fractal/blob/master/.travis.yml#L12

**Comment by GrahamCampbell** [comment]
Yeh, remove `--dev` from it too. It's deprecated, and is enabled by default.

**Comment by maximebeaudoin** [comment]
Humm, `The command "phpunit --coverage-text --coverage-clover=coverage.clover" exited with 255.` when travis try to build one PHP 5.3. Any idea ?

**Comment by GrahamCampbell** [comment]
Probably because it's scanning files with only php 5.4 valid syntax. :/

**Comment by maximebeaudoin** [comment]
This weird because i don't see any php 5.4 syntax in illuminate/contracts

**Comment by HipsterJazzbo** [comment]
What is left to be done before this can be merged? Is it just figuring out how to make travis do it's thing?

**Comment by maximebeaudoin** [comment]
@HipsterJazzbo Unfortunately, the package `illumintate/contracts` require php 5.4 and we can't run the test on travis php 5.3 environment because of this. So, i don't know how to deal with this at the moment.

**Comment by HipsterJazzbo** [comment]
Hmm. It's not really a matter of just getting travis to pass tests thought right? Like, if a dependency has 5.4 requirement, you won't be able to use it on 5.3 anyway? 

In that case, you either have to bump the PHP version requirement, or get rid of the dependency, right?

**Comment by GrahamCampbell** [comment]
I think we should just drop php 5.3 support totally. It's eol anyway. That would simplify this.

// cc @philsturgeon

**Comment by philsturgeon** [comment]
Seems fair. I was going to keep dropping 5.3 for 1.0 but lets just knock the old Laravel adapter out and go with this new one. 

Sorry to flip flop.

**Comment by GrahamCampbell** [comment]
@philsturgeon Sounds like a plan. @maximebeaudoin Are you able to update your pull request, or should I send a new one?

**Comment by maximebeaudoin** [comment]
@GrahamCampbell I'll update mine in few minutes.

**Comment by GrahamCampbell** [comment]
Great. :)

File: composer.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
         "phpunit/phpunit": "~4.0",
         "mockery/mockery": "~0.9",
         "illuminate/pagination": "~4.1",
```
Comment (GrahamCampbell) [inline composer.json]: Need to make this ~5.0.

Comment (GrahamCampbell) [inline composer.json]: Or even remove it if the contracts repo does the job.

File: .travis.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ php:
 
 before_script:
   - travis_retry composer self-update
```
Comment (GrahamCampbell) [inline .travis.yml]: We can remove this self-update call also.

File: .travis.yml [hunk 3]
```diff
@@ -9,7 +9,7 @@ php:
 
 before_script:
```
Comment (GrahamCampbell) [inline .travis.yml]: And change `before_script` to `install`.

File: composer.json (hunk 1 above)
Comment (maximebeaudoin) [inline composer.json]: I'll remove "illuminate/pagination".

Comment (GrahamCampbell) [inline composer.json]: The suggests block probably need updating too.

Comment (GrahamCampbell) [inline composer.json]: tbh, i think we should still "suggest" the pagination component itself.

Comment (maximebeaudoin) [inline composer.json]: Yep, i'll keep it

**Comment by GrahamCampbell** [comment]
I don't think we should rename the paginator class btw.

**Comment by maximebeaudoin** [comment]
@GrahamCampbell Yea, i'm wondering too because we have two different type of paginator into Laravel 5, but we can only support `LengthAwarePaginator` because some method implementation missing in default paginator

**Comment by philsturgeon** [comment]
Yeah im ok with just supporting length aware. It sounds like the other one would fit in as a cursor better anyway

**Comment by maximebeaudoin** [comment]
So i keep the name `IlluminatePaginatorAdapter`or `IlluminateLengthAwarePaginatorAdapter` ?

**Comment by GrahamCampbell** [comment]
IlluminatePaginatorAdapter is better imo. I see no reason to change it. The old one was length aware too.

**Comment by maximebeaudoin** [comment]
I think this is it !

**Comment by GrahamCampbell** [comment]
You need to bump the minor version in the branch alias too since this is a breaking changes.

**Comment by GrahamCampbell** [comment]
Otherwise, :+1:.

**Comment by philsturgeon** [comment]
IlluminatePaginatorAdapteror please

**Comment by philsturgeon** [comment]
Sorry my email client is f**ked up.

**Comment by philsturgeon** [comment]
I'll do that. Thank you!

**Pull request merged**